### ISSUE TITLE
[Service Bus Client] Migrate Outstanding Changes in the Dedicated Repository

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#5872]
# Summary

Since the initial migration of the Service Bus client library to the Azure SDK central repository, the service team has continued to make some changes in the legacy repository.   These changes should be captured and migrated to the central repository as part of finalizing the move.

# Scope of Work

- Capture changes to the Service Bus client library since the initial code drop was taken for migration and apply those changes to the client library source in the central repository.  This includes changes to the service and associated tests.

- Perform any needed restructuring or enhancements to the test code, to conform to the new patterns introduced in the central repository post-migration.

- Ensure that Live tests continue to pass consistently across all target platforms, both using local runs and the nightly builds from the `Internal` pipeline.

# Out of Scope

- Maintaining history of the recent commits; the changes will be applied to the central repository as a new change set, without an attempt to retain the original context.

- Enhancements, new features, fixes for currently existing issues, or formatting updates to the Service Bus client library; a key goal of the migration is to introduce as few changes as possible.

# Success Criteria

The changes made to the Service Bus client library since the initial migration to the central SDK repository have been migrated.   The service continues to build successfully; test runs for Live tests behave consistently and continue to pass.

# Details

- Changes should be identified in the  'dev` branch of the [legacy Service Bus repository](https://github.com/Azure/azure-service-bus-dotnet)  

- The commit used for the initial migration was [`b70ddf652c4c520fc127407be209b0811bcc50ce`](https://github.com/Azure/azure-service-bus-dotnet/commit/b70ddf652c4c520fc127407be209b0811bcc50ce) by Sean Feldman, on February 15, 2019.

- Changes should not include rolling back dependency versions nor introducing loose supporting assets from the central repository that were removed.

**Comment by SeanFeldman** [comment]
@jsquire please consider #5921 for this issue

**Comment by jsquire** [comment]
> @jsquire please consider #5921 for this issue

Hi @SeanFeldman.   A bit of context around the state of things:

The original migration attempted to preserve history as a best effort, bringing the unrelated histories forward as part of the migration using the technique that you referenced in #5921.   Unfortunately, trying to manage a 300+ commit change set over a period of time is a challenge.  Due to churn in the repository, structural changes, and other parallel work streams, there were several rebases needed to resolve conflicts.  Of those, there were a couple which were difficult, one I believe to have caused separation of the file history, though the [history of commits](https://github.com/Azure/azure-sdk-for-net/commits/master?after=5978dd5ffa3b26b0b2fc2c818ac98233459bc1a7+244) continues to exist as a parent chain to the [migration commit](https://github.com/Azure/azure-sdk-for-net/commit/d10ea6e0d90c55cdd1f97e8a6d1a126f67efc52d).  

We discussed post-migration, and it was decided that there was not enough value in the history that the efforts needed to preserve it were justified for future migrations.  Given our team's mission to create a unified and cohesive set of SDKs across Azure services, it is expected that many of the existing client libraries will undergo quite a bit of change in the near future, further limiting the long-term value of preserving history for the initial migration.

The current plan, as I understand it, is that the original repository will exist as an archive and represent the history pre-migration, and the point of migration into the central repository will constitute a new line of development.

With respect to cherry-picking this particular set of changes, my goal is to address the drift that has occurred between the repositories, as there have been changes in both, with the changes in the original repository being more tightly scoped and less extensive.  This isn't a case of replaying the changes on top of the current state,  I manually brought each over and made any needed adjustments to preserve that change in context of the current state of the new repository.   

Not the answer that you were looking for, I know, but hopefully it provides a bit of insight.

**Comment by SeanFeldman** [comment]
It would be easier to merge the original repo into SDKs repo as a single "cut over" commit, but when there are multiple steps, destination is changing while the source is undergoing modification, this will be almost impossible. Thank you for the context.

**Comment by jsquire** [comment]
Agreed.  One of the challenges of migrating to a monorepo using a shared engineering system the potential for breaking the build and cascading failures to unrelated areas.  Ours is one of the moving parts under active development and doesn't currently have a means of ignoring a particular area or path.  Whatever is merged in must be compatible.  While we had the commit history as a parent chain when we started, we had to make the changes to adapt to the engineering system and repository structure, and ensure that the test suite was stable enough before we could integrate it.

In hindsight, I'd say that trying to maintain a linear history was a mistake and agree that it would have been a better approach - and easier - to have a merge commit as a starting point.  We also didn't anticipate that there would be contention for the destination root directory from the management library that was being restructured in parallel.  There were most certainly some missteps and lessons learned.
